src/fcc_bridge_node_ros.cpp

```diff
@@ -0,0 +1,506 @@
+    200}; /**< The period of the 5Hz telemetry timer */
+constexpr std::chrono::milliseconds FCC_TELEMETRY_PERIOD_10HZ{
+    100}; /**< The period of the 10Hz telemetry timer */
+
+// Mission control node name
+constexpr char const *const MISSION_CONTROL_NODE_NAME = "mission_control";
+
+// Limits
+constexpr std::chrono::seconds MAX_UAV_COMMAND_AGE{
+    1}; /**< Maximum age of a received UAV command */
+
+}  // namespace
+
+void FCCBridgeNode::set_internal_state(
+    const fcc_bridge::FCCBridgeNode::INTERNAL_STATE new_state) {
+    // Actually set the state.
+    this->internal_state = new_state;
+
+    RCLCPP_INFO(this->get_logger(), "Switching to new internal_state: %s",
+                this->internal_state_to_str());
+}
+
+void FCCBridgeNode::setup_ros() {
+    RCLCPP_DEBUG(this->get_logger(), "Setting up ROS");
+    // Check if the node is in the correct state
+    if (this->get_internal_state() != INTERNAL_STATE::STARTING_UP) {
+        RCLCPP_ERROR(this->get_logger(),
+                     "Attempted to setup ROS components more than once");
+        this->set_internal_state(INTERNAL_STATE::ERROR);
+        return;
+    }
+
+    // Setup publisher
+
+    // Create GPSInfo & Position publisher
+    this->gps_position_publisher =
+        this->create_publisher<interfaces::msg::GPSPosition>(
+            common_lib::topic_names::GPSPosition, 1);
+    // Create FlightState publisher
+    this->flight_state_publisher =
+        this->create_publisher<interfaces::msg::FlightState>(
+            common_lib::topic_names::FlightState, 1);
+
+    // Create BatteryState publisher
+    this->battery_state_publisher =
+        this->create_publisher<interfaces::msg::BatteryState>(
+            common_lib::topic_names::BatteryState, 1);
+
+    // Create RCState publisher
+    this->rc_state_publisher = this->create_publisher<interfaces::msg::RCState>(
+        common_lib::topic_names::RCState, 1);
+
+    // Create euler angle publisher
+    this->euler_angle_publisher = this->create_publisher<interfaces::msg::Pose>(
+        common_lib::topic_names::Pose, 1);
+
+    // Create mission progress publisher
+    this->mission_progress_publisher =
+        this->create_publisher<interfaces::msg::MissionProgress>(
+            common_lib::topic_names::MissionProgress, 1);
+
+    // Create UAV health publisher
+    this->uav_health_publisher =
+        this->create_publisher<interfaces::msg::UAVHealth>(
+            common_lib::topic_names::UAVHealth, 1);
+
+    // Create mission start publisher
+    this->mission_start_publisher =
+        this->create_publisher<interfaces::msg::MissionStart>(
+            common_lib::topic_names::MissionStart, 1);
+
+    // Setup subscriber
+
+    // Create subscription options for heartbeat to only receive mission control
+    // heartbeats
+    rclcpp::SubscriptionOptions subscription_options;
+    subscription_options.content_filter_options.filter_expression =
+        "sender_id = 'mission_control'";

```

Comment:
Maybe use your constant for the mission control node name here

Reply:
Fixed in 2734907